src/evd.cpp

```diff
@@ -19,3 +19,90 @@ void sym_schur2(const double* const X, int n, int p, int q, double* c, double* s
         *s = 0.0;
     }
 }
+
+void evd_classic_tol(struct matrix_t Xmat, struct vector_t evec, struct matrix_t Qmat, double tol) {
+    const int n = Xmat.cols;
+    double* X = Xmat.ptr;
+    double* e = evec.ptr;
+    double* Q = Qmat.ptr;
+    //A=QtXQ
+    double *A = (double*)malloc(sizeof(double)*n*n);
+
+    double offA=0, abs_a, eps, c, s;
+    int p, q;
+
+    for(int i = 0; i < n; ++i){
+        for(int j = 0; j<n; ++j){
+            A[n*i+j] = X[n*i+j];
+        }
+    }
+    for(int i = 0; i < n; ++i){
+        Q[n*i+i] = 1.0;
+    }
+    for(int i = 0; i < n; ++i){
+        for(int j = i+1; j<n; ++j){
+            double a_ij = A[n*i+j];
+            offA+=2*a_ij*a_ij;
+        }
+    }
+    for(int i = 0; i < n; ++i){
+        for(int j = i; j<n; ++j){
+            double a_ij = A[n*i+j];
+            if(i == j){
+                eps+=a_ij*a_ij;
+            } else {
+                eps+=2*a_ij*a_ij;
+            }
+        }
+    }
+    eps = tol*tol*eps;
+
+    while(offA > eps){
+
+        abs_a = 0.0;
+        for(int i = 0; i < n; ++i){
+            for(int j = i+1; j<n; ++j){
+                double abs_ij = abs(A[i*n+j]);
+                if(abs_ij > abs_a){
+                    abs_a = abs_ij;
+                    p = i;
+                    q = j;
+                }
+            }
+        }
+
+        sym_schur2(A,n,p,q,&c,&s);
```

Comment:
Could you use the new ```jacobi_sym_coeffs``` function here? They have the same functionality, but the new one is more generic. We can delete this function at a later stage.

Reply:
will do!